test/sharness/t0087-repo-robust-gc.sh

```diff
@@ -0,0 +1,143 @@
+#!/bin/sh
+#
+# Copyright (c) 2016 Jeromy Johnson
+# MIT Licensed; see the LICENSE file in this repository.
+#
+
+test_description="Test ipfs pinning operations"
+
+. lib/test-lib.sh
+set -e
+
+test_gc_robust_part1() {
+
+	test_expect_success "add a 1MB file with --raw-leaves" '
+		random 1048576 56 > afile &&
+		HASH1=`ipfs add --raw-leaves -q afile`
+	'
+
+	LEAF1=zb2rhZWKCMcHCfCym41ocwE5BhNQsEVLrwBkLEQaxJzRHHZch
+	LEAF1FILE=.ipfs/blocks/C2/AFKREIBKRCW7HF6NHR6DVWECQXC5RQC7U7PKHKTI53BYYZNQP23DK5FC2Y.data
+
+	LEAF2=zb2rhdUpCuyoQPRtvBW8JhAEfPaNd9dBb3QQCMMWYS1V8hjBR
+	LEAF2FILE=.ipfs/blocks/Q6/BAFKREIDFSUIR43GJPHNDXXQA45GJVNRZBET3CRPUMYJCBLK3RTN7ZAMQ6Q
+
+	LEAF3=zb2rhnwyvreekPqEMAbKqs35KYHxkNi7pXED7L2TfJKveTTva
+	LEAF4=zb2rhnvAVvfDDnndcfQkwqNgUm94ba3zBSXyJKCfVXwU4FXx
+
+	test_expect_success "remove a leaf node from the repo manually" '
+		rm $LEAF1FILE
+	'
+
+	test_expect_success "check that the node is removed" '
+		test_must_fail ipfs cat $HASH1
+	'
+
+	test_expect_success "'ipfs repo gc' should still be be fine" '
+		ipfs repo gc
+	'
+
+	test_expect_success "unpin the 1MB file" '
+		ipfs pin rm $HASH1
+	'
+
+	# make sure the permission problem is fixed on exit, otherwise cleanup
+	# will fail
+	trap "chmod 700 `dirname $LEAF2FILE` 2> /dev/null" 0
+
+	test_expect_success "create a permission problem" '
+		chmod 500 `dirname $LEAF2FILE` &&
+		test_must_fail ipfs block rm $LEAF2 2>&1 | tee err &&
```

Comment:
So the reason I do it this was is so rather than a simple '2> err' is so that I can see the output of the command when developing the test.  It is far easier than having to do a `cat "trash directory XXX/err"`.  The sharness framework will suppress both stdout and stderr from a test unless '-v' is given so I don't see extra noise as a problem.  If the test does fail for some reason then it is helpful to see the output of the command without having to track down the output file name.  If a test fails on a build server the output file may not even be accessible.

Nevertheless, I understand it is not the convention used in the other tests, so if you still don't like I it I will reluctantly change it after I am done developing the tests and the code.  Let me know.